### ISSUE TITLE
[policy] catch policy

### ISSUE THREAD

**Issue by vdelattre** [gravitee-io/issues#793]
it could be useful to have a policy to catch error from policies or backend and be able to override return code or message.

example:
1 - api with 1 policy OAuth and 1 policy JWT. Be able to now which policy failed. 
2 - healthckeck has disable all backend, return is 503 Service Unavailable, it could be nice to can add custom message.

discussed with david

### Your Environment
* Version used: 1.9.0

**Comment by Matsa59** [comment]
I agree with you @vdelattre , I saw a lot of APIs that returned 200 even if the behavior described an error.

E.g. you got a response like that :
Response status header: 200
```json
{
    "error": "...",
    "error_description": "..."
}
```

It could be really nice if the Catch Policy is able to determine another HTTP status header depending of the backend response content. Of course, we should be able to use backend response body to define the gravitee response body.

Are you agree @vdelattre ?
